nntrainer/optimizers/optimizer_impl.h

```diff
@@ -16,10 +16,44 @@
 #define __OPTIMIZER_IMPL_H__
 #ifdef __cplusplus
 
+#include <tuple>
+
+#include <base_properties.h>
 #include <optimizer_devel.h>
 
 namespace nntrainer {
 
+/**
+ * @brief Learning Rate props
+ *
+ */
+class PropsLR : public Property<float> {
+public:
+  static constexpr const char *key =
+    "learning_rate";               /**< unique key to access */
+  using prop_tag = float_prop_tag; /**< property type */
+};
+
+/**
+ * @brief Decay rate property
+ *
+ */
+class PropsDecayRate : public Property<float> {
+public:
+  static constexpr const char *key = "decay_rate"; /**< unique key to access */
+  using prop_tag = float_prop_tag;                 /**< property type */
+};
+
+/**
+ * @brief decay steps property
+ *
+ */
+class PropsDecaySteps : public PositiveIntegerProperty {
```

Comment:
decay steps must be more than 0 if defined?

Reply:
Yes, now we have `PositiveIntegerProperty` as suggested earlier :)